src/Features/Core/Portable/ConvertNumericLiteral/AbstractConvertNumericLiteralCodeRefactoringProvider.cs

```diff
@@ -19,26 +19,22 @@ internal abstract class AbstractConvertNumericLiteralCodeRefactoringProvider : C
         {
             var document = context.Document;
             var cancellationToken = context.CancellationToken;
+            var syntaxFacts = document.GetLanguageService<ISyntaxFactsService>();
             var root = await document.GetSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
+            var numericToken = await root.SyntaxTree.GetTouchingTokenAsync(context.Span.Start,
+                token => syntaxFacts.IsNumericLiteralExpression(token.Parent), cancellationToken).ConfigureAwait(false);
 
-            var numericToken = root.FindToken(context.Span.Start);
-            if (numericToken.ContainsDiagnostics)
+            if (numericToken == default(SyntaxToken))
             {
                 return;
             }
 
-            if (!numericToken.Span.Contains(context.Span))
```

Comment:
This means the refactoring will now trigger if there is a selection that *starts* in the literal.  I think I'd prefer that it only work if there is either:

* No selection
* The selection matches Token.Span.

Thoughts?